.pre-commit-config.yaml

```diff
@@ -0,0 +1,30 @@
+exclude: "docs"
+
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.3.0
+    hooks:
+      - id: trailing-whitespace
+      - id: end-of-file-fixer
+
+  - repo: https://github.com/asottile/pyupgrade
+    rev: v2.34.0
+    hooks:
+      -  id: pyupgrade
+         args: [--py37-plus]
+
+  - repo: https://github.com/psf/black
+    rev: 22.3.0
+    hooks:
+      - id: black
```

Comment:
Hi @addisonElliott thanks for reviewing the PR. If I run `pylint` instead of `black`,  there is a large amount of things non-compliant. Would you prefer me to switch to `pylint` and fix all the things?
Examples:
```
....
************* Module nrrd
nrrd/__init__.py:1:0: C0114: Missing module docstring (missing-module-docstring)
************* Module nrrd.errors
nrrd/errors.py:1:0: C0114: Missing module docstring (missing-module-docstring)
nrrd/errors.py:4:4: W0107: Unnecessary pass statement (unnecessary-pass)
************* Module nrrd.formatters
nrrd/formatters.py:1:0: C0114: Missing module docstring (missing-module-docstring)
nrrd/formatters.py:1:0: E0401: Unable to import 'numpy' (import-error)
nrrd/formatters.py:4:18: C0103: Argument name "x" doesn't conform to snake_case naming style (invalid-name)
nrrd/formatters.py:41:18: C0103: Argument name "x" doesn't conform to snake_case naming style (invalid-name)
nrrd/formatters.py:60:27: C0103: Argument name "x" doesn't conform to snake_case naming style (invalid-name)
nrrd/formatters.py:81:4: R1705: Unnecessary "else" after "return", remove the "else" and de-indent the code inside it (no-else-return)
nrrd/formatters.py:87:18: C0103: Argument name "x" doesn't conform to snake_case naming style (invalid-name)
nrrd/formatters.py:106:27: C0103: Argument name "x" doesn't conform to snake_case naming style (invalid-name)
nrrd/formatters.py:132:23: C0103: Argument name "x" doesn't conform to snake_case naming style (invalid-name)
....
```

Do we want to run `pylint` and other hooks in the test files as well? The list of errors reported by Pylint is very large.

Looking forward to hearing your thoughts!

Reply:
`pylint` can be a bit rough too since it's very verbose. It takes a lot of configuring to get a good setup going.

What about no `black` and no `pylint` and start with flake8?

Reply:
sounds good, fixing it

Reply:
ready for review. Removed black